libtenzir/builtins/contexts/lookup_table.cpp

```diff
@@ -247,27 +256,55 @@ class ctx final : public virtual context {
       }
       return std::nullopt;
     };
-    for (const auto& value : array.values()) {
+    auto now = time::clock::now();
+    for (auto value : array.values()) {
       if (auto it = context_entries.find(materialize(value));
           it != context_entries.end()) {
-        builder.data(it->second);
+        if ((it->second.create_timeout && it->second.create_timeout < now)
+            || (it->second.update_timeout && it->second.update_timeout < now)) {
+          continue;
+        }
+        if (it->second.update_timeout) {
+          *it.value().update_timeout = now + update_time_;
+        }
```

Comment:
This you only want to do when you had a successful lookup for either subnets or the regular table.